sample/weighted.go

```diff
@@ -0,0 +1,118 @@
+// Copyright ©2015 The gonum Authors. All rights reserved.
+// Use of this code is governed by a BSD-style
+// license that can be found in the LICENSE file
+
+package sample
+
+import (
+	"math/rand"
+
+	"github.com/gonum/floats"
+)
+
+// Weighted provides sampling without replacement from a collection of items with
+// non-uniform probability.
+type Weighted struct {
+	weights []float64
+	// heap is a weight heap. See comments
```

Comment:
I don't think this is sufficiently descriptive. Sorry. 

container/heap says "A heap is a tree with the property that each node is the minimum-valued node in its subtree". That's misleading for a number of reasons. 1) Our sign is flipped, ours is higher (okay, whatever). 2) Both of the examples in container heap are for keeping a sorted list of values. Our heap does not maintain a sorted list of weights, nodes contain sums of subsets of weights. 3) The public documentation for container/heap does not say anything about traversing, which is needed for reading this code. At the least it should point to the implementation (see code for up/down in container heap). 4) Related to 2, the code in container/heap does not look like this code. It makes calls to Less and Swap, but there are no operations that look like Less and Swap in the code to reset. 

It's okay to use a pointer to container/heap to describe what a heap is, but should be a description of what the data in this heap actually is.